src/Propel/Common/Config/Loader/FileLoader.php

```diff
@@ -61,6 +61,8 @@ public function __construct(FileLocatorInterface $locator = null)
      * Replaces parameter placeholders (%name%) by their values for all parameters.
      *
      * @param array $configuration The configuration array to resolve
+     *
+     * @return array|void
```

Comment:
I still disagree on that and what you claim in the blog post. Semantically, you can not just replace void by null just because the function has a state where it can return something not null'ish. The interpretation of `array|void` is different to `array|null` since later would imply null is signification as its a actual type. That is completely independent of the fact that you check the result type.

However, one thing is for sure: Changing that annotation means changing the function behaviour and its interpretation, which is out of scope of this PR. So, technically, its completely fine to place void here, since `return;` is void.  Whether that makes sense in combination with another return type is a question you can ask later on - although I'd still disagree in your interpretation of what `void` means in combination with a real type.